client/rest-high-level/src/main/java/org/elasticsearch/client/IndicesRequestConverters.java

```diff
@@ -97,7 +97,22 @@ static Request closeIndex(CloseIndexRequest closeIndexRequest) {
         return request;
     }
 
-    static Request createIndex(CreateIndexRequest createIndexRequest) throws IOException {
+    static Request createIndex(CreateIndexRequest createIndexRequest)
+        throws IOException {
+        String endpoint = RequestConverters.endpoint(createIndexRequest.indices());
+        Request request = new Request(HttpPut.METHOD_NAME, endpoint);
+
+        RequestConverters.Params parameters = new RequestConverters.Params(request);
+        parameters.withTimeout(createIndexRequest.timeout());
+        parameters.withMasterTimeout(createIndexRequest.masterNodeTimeout());
+        parameters.withWaitForActiveShards(createIndexRequest.waitForActiveShards());
+
+        request.setEntity(RequestConverters.createEntity(createIndexRequest, RequestConverters.REQUEST_BODY_CONTENT_TYPE));
+        return request;
+    }
+
+    static Request createIndex(org.elasticsearch.action.admin.indices.create.CreateIndexRequest createIndexRequest)
```

Comment:
Maybe mark as deprecated, even if its only used internally.

Reply:
I haven't been adding deprecation annotations to our internal code, especially since we can't actually move off of these methods (we need to continue to support this functionality for now). So I'd be inclined to not add a warning here for consistency, does that make sense to you as well?

Reply:
We should be deleting this in 8.0 right? Im fine w/ leaving it as is assuming a followup PR will be nuking it from orbit once we cut 7.x branches. Your call.

Reply:
Right, we'll be removing this in 8.0. I'll plan to keep it as is for this PR.